src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -2,46 +2,50 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.IO;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Html;
 using Microsoft.Extensions.Internal;
 
 namespace Microsoft.AspNetCore.Razor.TagHelpers
 {
     /// <summary>
     /// An HTML tag helper attribute.
     /// </summary>
-    public class TagHelperAttribute
+    public class TagHelperAttribute : IHtmlContentContainer
     {
         /// <summary>
         /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>.
-        /// <see cref="Minimized"/> is set to <c>true</c> and <see cref="Value"/> to <c>null</c>.
+        /// <see cref="Structure"/> is set to <see cref="HtmlAttributeStructure.Minimized"/> and <see cref="Value"/> to
+        /// <c>null</c>.
         /// </summary>
         /// <param name="name">The <see cref="Name"/> of the attribute.</param>
         public TagHelperAttribute(string name)
-            : this(name, value: null, minimized: true)
+            : this(name, value: null, structure: HtmlAttributeStructure.Minimized)
         {
         }
 
         /// <summary>
         /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>
-        /// and <paramref name="value"/>. <see cref="Minimized"/> is set to <c>false</c>.
+        /// and <paramref name="value"/>. <see cref="Structure"/> is set to <see cref="HtmlAttributeStructure.DoubleQuotedValue"/>.
         /// </summary>
         /// <param name="name">The <see cref="Name"/> of the attribute.</param>
         /// <param name="value">The <see cref="Value"/> of the attribute.</param>
         public TagHelperAttribute(string name, object value)
```

Comment:
Suggest we break compat, remove this method, and force users to think about the structure -- especially if their `value` includes `HtmlString`s. Thoughts @NTaylorMullen @Eilon?